### ISSUE TITLE
Bad error resolve message

### ISSUE THREAD

**Issue by kostya** [crystal-lang/shards#313]
Crystal 0.32.1 (2019-12-18)
Shards 0.8.1 (2019-12-18)

```yaml
dependencies:
  db:
    github: crystal-lang/crystal-db
  pg:
    github: will/crystal-pg
  crecto:
    github: fridgerator/crecto
```

```
shards
Fetching https://github.com/crystal-lang/crystal-db.git
Fetching https://github.com/will/crystal-pg.git
Fetching https://github.com/fridgerator/crecto.git
Error resolving db (*, ~> 0.8.0, ~> 0.7.0)
```

hard to understand whats wrong here, who depend on what.

**Comment by bcardiff** [comment]
I agree. FWIW latest crecto released 2 months ago uses crystal-db 0.7.0

**Comment by ysbaddaden** [comment]
Shards' resolver prior to 0.9 is far too stupid, and barely capable to detect conflicts (not always). Analyzing and reporting them is impossible in v0.8.
